### ISSUE TITLE
Sort files by depth when loading autoload dir

### ISSUE THREAD

**Issue by andreyorst** [mawww/kakoune#2808]
https://github.com/mawww/kakoune/blob/master/share/kak/kakrc#L28-L49

Currently `autoload` folder is sourced in order which `find` command returns, which may lead to issues like so: https://github.com/andreyorst/fzf.kak/issues/45

While I understand that there's currently no way to tell one script that it relies on another, I've tried to make module like system where modules that depend on some file are located deeper in the tree:

```
rc/
  main.kak
  modules/
    module1.kak
    module2.kak
  misk.kak
```

`module1.kak` and `module2.kak` are closely related to `main.kak` and can't work without it, while `misk.kak` is not, and can be sourced any time.

My plugin manager handles this, by sorting `find` output by depth, using [perl sort function](https://github.com/andreyorst/plug.kak/blob/fe165c04d259983021da41faa2809de3fec8379b/rc/plug.kak#L173-L177).

I don't know, but maybe Kakoune should follow the same pattern for organizing modules?

**Comment by laelath** [comment]
Would this also be solved by using the module system #2782, and putting the loading in a KakBegin hook?

**Comment by andreyorst** [comment]
> Would this also be solved by using the module system #2782, and putting the loading in a KakBegin hook?

didn't diven to this system yet, but probably yes.
